Issue 21220: improve error handling for botEnableCmdF function

ariyonaty opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->
Error handling for the `botEnableCmdF` function updated to return an error in case of failure. 
Update includes the use of `multierror.Error` for appending all errors and returning error if one occurs.

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/21220

Otherwise, link the JIRA ticket.
-->

Fixes https://github.com/mattermost/mattermost-server/issues/21220

mattermod commented:
Hello @ariyonaty,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

isacikgoz reviewed (approved):
Thanks for the contribution @ariyonaty LGTM!

noxer commented on commands/bot.go:
```diff
@@ -216,22 +217,27 @@ func botListCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	for i, user := range users {
 		if user == nil {
 			printer.PrintError(fmt.Sprintf("can't find user '%v'", args[i]))
+			result = multierror.Append(result, fmt.Errorf("can't find user '%v'", args[i]))
```
Please use `%q` to quote strings
```suggestion
			result = multierror.Append(result, fmt.Errorf("can't find user %q", args[i]))
```

noxer commented on commands/bot.go:
```diff
@@ -216,22 +217,27 @@ func botListCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 		bot, _, err := c.EnableBot(user.Id)
 		if err != nil {
 			printer.PrintError(fmt.Sprintf("could not enable bot '%v'", args[i]))
+			result = multierror.Append(result, fmt.Errorf("could not enable bot '%v'", args[i]))
```
Same here. Please also include the error with `%w`.
```suggestion
			result = multierror.Append(result, fmt.Errorf("could not enable bot %q: %w", args[i], err))
```

noxer commented on commands/bot_e2e_test.go:
```diff
@@ -198,7 +198,7 @@ func (s *MmctlE2ETestSuite) TestBotEnableCmd() {
 
 		err := botEnableCmdF(s.th.Client, &cobra.Command{}, []string{newBot.UserId})
-		s.Require().Nil(err)
+		s.Require().NotNil(err)
```
Please check this with `s.Require().Error(err)`, it makes the test messages much easier to read.

noxer commented on commands/bot_e2e_test.go:
```diff
@@ -209,7 +209,7 @@ func (s *MmctlE2ETestSuite) TestBotEnableCmd() {
 
 		err := botEnableCmdF(c, &cobra.Command{}, []string{"nonexistent-bot-userid"})
-		s.Require().Nil(err)
+		s.Require().NotNil(err)
```
Same as above

noxer reviewed (changes requested):
Hi @ariyonaty, thank you very much for your contribution. I've got a few requests but otherwise it looks good.

noxer commented:
Hey @ariyonaty, do you have questions about the proposed changes?

ariyonaty commented:
@noxer - Appreciate the feedback.  
Addressed the comments in commit [5c1a6fb](https://github.com/mattermost/mmctl/pull/577/commits/5c1a6fbf5af85b37574df6ccdafc72750e1f37ec).

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

noxer reviewed (approved):
Looks good, thank you very much @ariyonaty. :rocket:
